### DISCUSSION TITLE
fatal error: SPI.h: No such file or directory

### DISCUSSION THREAD

**Discussion by gorillale** [olikraus/u8g2#1533] [General]
- this is fixed with reinstal the bard => fatal error: SPI.h: No such file or directory

The other problem is the error:
undefined reference to u8x8_SetPin_HW_I2C

```
U8G2_SSD1306_128X64_NONAME_F_HW_I2C* pDsp;

setup() {
	pDsp = new U8G2_SSD1306_128X64_NONAME_F_HW_I2C(U8G2_R0, /* reset=*/ U8X8_PIN_NONE, SCL, SDA);
	pDsp->setI2CAddress(iLcdAdr);
	vDsp->init(pDsp, 0); 
}

```
On myClass vDsp is another pointer, get with vDsp->init(...)
With other Display Driver i have not this problem, class works without errors.

What can i do to find a solution to eliminate this error?

**Comment by olikraus** [comment]
Looks like u8x8lib.cpp is missing

**Reply by gorillale** [reply]
Hi,
no, i have included, and same error.
I have make a minimal code. a empty class.cpp , then i add another #include <u82glib.h> into header, the error displays.
when i remove from main.ino other errors occures.

with one ino file and one include library u82g works

**Reply by gorillale** [reply]
I found the mistake: the class include must be the last after all includes....
not known why...

and i see on your library:
i think it is not good idea to enable SPI at all on first include into your library and then disable at used display, 
i must include the spi allways but never used.... 
sorry. problems with esp8266 boards
